src/Shared/CommunicationsUtilities.cs

```diff
@@ -387,14 +387,14 @@ internal static void WriteLongForHandshake(this PipeStream stream, long value)
         /// </summary>
         internal static long ReadLongForHandshake(this PipeStream stream, byte[] leadingBytesToReject,
             byte rejectionByteToReturn
-#if NETCOREAPP2_1
+#if NETCOREAPP2_1 || MONO
             , int timeout
 #endif
             )
         {
             byte[] bytes = new byte[8];
 
-#if NETCOREAPP2_1
+#if NETCOREAPP2_1 || MONO
             if (!NativeMethodsShared.IsWindows)
```

Comment:
Out of curiosity:
1. Why was this enabled only for Core 2.1 and not for all non-Framework builds?
2. Why don't we use a/the timeout in the else branch below? It looks like `stream.ReadByte()` may also hang forever if the node is broken.

Reply:
1. We're pretty inconsistent across the codebase with our ifdefs. This was (*checks history to make sure it was me*) probably just sloppiness and forgetting about Mono. Unfortunately, since Mono presents as Framework, we can't just do `#if !NETFRAMEWORK`
2. I think that was mostly "don't change legacy code without good reason" + "no async or timeout-enabled `PipeStream.ReadByte()`. But I don't really remember :(